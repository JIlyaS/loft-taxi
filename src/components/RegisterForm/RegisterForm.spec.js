import React from 'react';
import {RegisterForm} from './';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { MuiPickersUtilsProvider } from '@material-ui/pickers';
import DateFnsUtils from '@date-io/date-fns';

import {Provider} from 'react-redux';
import createStore from '../../store';

const store = createStore();

const renderWithRouter = (component, {route='/', history = createMemoryHistory({ initialEntries: [route]})} = {}) => {
  const Wrapper = ({ children }) => (
  <Provider store={store}>
    <Router history={history}>
        <MuiPickersUtilsProvider utils={DateFnsUtils}>
          { children }
        </MuiPickersUtilsProvider>
      </Router>
    </Provider>
  );

  return {
    ...render(component, {wrapper: Wrapper}),
    history,
  };
};

describe('RegisterForm', () => {
  it('renders correctly spanshot', () => {
    const history = createMemoryHistory();
    const tree = renderer.create(<Provider store={store}>
      <Router history={history}>
        <MuiPickersUtilsProvider utils={DateFnsUtils}>
          <RegisterForm fetchRegisterRequestAction={jest.fn()} />
        </MuiPickersUtilsProvider>
      </Router>
    </Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correct', () => {
    const { getByTestId } = renderWithRouter(<RegisterForm fetchRegisterRequestAction={jest.fn()} />);
    const registerForm = getByTestId('register-form');
    expect(registerForm).toBeInTheDocument();
  });

  it('Change user email successfully', () => {
    renderWithRouter(<RegisterForm fetchRegisterRequestAction={jest.fn()} />);
    const input = screen.getByPlaceholderText('mail@mail.ru');
    expect(input).toBeInTheDocument();
    fireEvent.change(input, { target: { value: 'ilkolmakov@mail.ru' } });
    expect(input.value).toBe('ilkolmakov@mail.ru');
  });

  it('Change user name successfully', () => {
    renderWithRouter(<RegisterForm fetchRegisterRequestAction={jest.fn()} />);
    const input = screen.getByPlaceholderText('Петр Александрович');
    expect(input).toBeInTheDocument();
    fireEvent.change(input, { target: { value: 'Иван Иванов' } });
    expect(input.value).toBe('Иван Иванов');
  });
    
  it('Change date input successfully', () => {
    renderWithRouter(<RegisterForm fetchRegisterRequestAction={jest.fn()} />);
    const input = screen.getByPlaceholderText('************');
    expect(input).toBeInTheDocument();
    fireEvent.change(input, { target: { value: '123' } });
    expect(input.value).toBe('123');
  });
});
