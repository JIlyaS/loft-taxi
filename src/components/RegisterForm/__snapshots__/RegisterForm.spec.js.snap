// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterForm renders correctly spanshot 1`] = `
<div
  className="register-form"
  data-testid="register-form"
>
  <form
    onSubmit={[Function]}
  >
    <h2
      className="register-form__title"
    >
      Регистрация
    </h2>
    <div
      className="input register-form__block"
    >
      <div
        className="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-1 MuiFormControl-marginDense MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense Mui-required Mui-required"
          data-shrink={false}
          htmlFor="email"
          id="email-label"
        >
          Email
          <span
            aria-hidden={true}
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense MuiInput-marginDense"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={true}
            className="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
            disabled={false}
            id="email"
            name="email"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="mail@mail.ru"
            required={true}
            type="email"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      className="input register-form__block"
    >
      <div
        className="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-1 MuiFormControl-marginDense MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense Mui-required Mui-required"
          data-shrink={false}
          htmlFor="name"
          id="name-label"
        >
          Как вас зовут?
          <span
            aria-hidden={true}
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense MuiInput-marginDense"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
            disabled={false}
            id="name"
            name="name"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Петр Александрович"
            required={true}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      className="input register-form__block"
    >
      <div
        className="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-1 MuiFormControl-marginDense MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense Mui-required Mui-required"
          data-shrink={false}
          htmlFor="password"
          id="password-label"
        >
          Пароль
          <span
            aria-hidden={true}
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense MuiInput-marginDense"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
            disabled={false}
            id="password"
            name="password"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="************"
            required={true}
            type="password"
            value=""
          />
        </div>
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root WithStyles(ForwardRef(Button))-root-2 MuiButton-contained register-form__login-btn MuiButton-containedPrimary MuiButton-disableElevation MuiButton-fullWidth"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      <span
        className="MuiButton-label"
      >
        Зарегистрироваться
      </span>
    </button>
    <div
      className="register-form__new-block"
    >
      <span
        className="register-form__new-text"
      >
        Уже зарегестрированны?
      </span>
      <a
        className="register-form__reg"
        href="/login"
        onClick={[Function]}
      >
        Войти
      </a>
    </div>
  </form>
</div>
`;
